packages/mobx-react/__tests__/observer.test.tsx

```diff
@@ -908,3 +908,46 @@ test("Missing render should throw", () => {
     }
     expect(() => observer(Component)).toThrow()
 })
+
+test("this.context is observable if ComponentName.contextType is set", () => {
+    const Context = createContext({})
+
+    class Parent extends React.Component<any> {
+        state = { counter: 0 }
+
+        componentDidMount() {
+            this.setState({ counter: 1 })
+        }
+
+        render() {
+            return <Context.Provider value={this.state}>{this.props.children}</Context.Provider>
+        }
+    }
+
+    @observer
+    class Child extends React.Component {
+        static contextType = Context
+
+        @computed
+        get counterValue() {
+            return (this.context as any).counter
+        }
+
+        render() {
+            return <div />
+        }
+    }
+
+    const compRef = React.createRef<Child>()
+
+    const { unmount } = render(
+        <Parent>
+            <Child ref={compRef} />
+        </Parent>
+    )
+
+    console.log(compRef.current?.counterValue)
+
+    expect(compRef.current?.counterValue).toBe(1)
```

Comment:
Have you tried to run the test without the fix? I think this would pass even before.

Reply:
This passes in both cases. But this why I used the store before. Like here https://github.com/mobxjs/mobx/blob/36d3d35047c3002376fca676d79399f381af7319/packages/mobx-react/__tests__/observer.test.tsx#L56

Reply:
A test that passes before and aft doesn't sound like a good test :) I can't see the original test clearly anymore, would you mind adding that as well for now at least? I do think atm this PR is correct, but that it isn't the "correct" solution to the original problem, from scanning the original issues 😅

Reply:
You can see the original test in this commit.

https://github.com/mobxjs/mobx/pull/3404/commits/4088edd197ef79f694e9245544f4dee48281cce5

Reply:
To clarify; in the original problem your 'store' isn't a store at at all, it is an unstable object reference for something that should be stable in the mobx idioms, and the correcter way to solve it would be simple: `<FormContext.Provider value={this}>` and skip that whole computed thing. Computed values shouldn't normally be new objects that are observable / mutable; either they are a ref to an already existing object, or they are immutable value objects, just representing the derived value at a certain moment in time. 

In your current setup you create many objects that might all live at the same moment in time, representing many different selections. 

Edit: fixed snippet

Reply:
Just use the computed value inside the render and it will fail. The computed doesn't cache if it's not observed, therefore the problem doesn't occur.

Reply:
Ah, for the first test, yeah that didn't seem to make sense indeed, that should pass even if context would be an observable.